src/libraries/Pool.sol

```diff
@@ -307,23 +300,22 @@ library Pool {
         Slot0 memory slot0Start = self.slot0;
         bool zeroForOne = params.zeroForOne;
 
-        SwapCache memory cache = SwapCache({
-            liquidityStart: self.liquidity,
-            protocolFee: zeroForOne ? slot0Start.protocolFee.getZeroForOneFee() : slot0Start.protocolFee.getOneForZeroFee()
-        });
+        uint128 liquidityStart = self.liquidity;
+        uint256 protocolFee =
+            zeroForOne ? slot0Start.protocolFee.getZeroForOneFee() : slot0Start.protocolFee.getOneForZeroFee();
 
         state.amountSpecifiedRemaining = params.amountSpecified;
         state.amountCalculated = 0;
         state.sqrtPriceX96 = slot0Start.sqrtPriceX96;
         state.tick = slot0Start.tick;
         state.feeGrowthGlobalX128 = zeroForOne ? self.feeGrowthGlobal0X128 : self.feeGrowthGlobal1X128;
-        state.liquidity = cache.liquidityStart;
+        state.liquidity = liquidityStart;
 
         // if the beforeSwap hook returned a valid fee override, use that as the LP fee, otherwise load from storage
-        uint24 lpFee =
+        slot0Start.lpFee =
```

Comment:
this is changing some functionality im surprised there isnt a failing test - the hook returned fee shouldnt be written to storage

Reply:
ohhh it isnt storage lol